internal/cmd/rootfs2vhd/rootfs2vhd.go

```diff
@@ -0,0 +1,187 @@
+package main
+
+// rootfs2vhd is a simple utility to convert rootfs.tar.gz generated by
+// the Microsoft/opengcs repos build.ps1 script to rootfs.vhd which LCOW
+// can use for the root filesystem added on VPMem (as opposed to an initrd).
+//
+// It's not pretty, but enough to get the job done. It does require
+// the Hyper-V RSAT to be installed for shelling out to New-VHD.
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/Microsoft/hcsshim/internal/schema2"
+	"github.com/Microsoft/hcsshim/internal/uvm"
+	"github.com/Microsoft/hcsshim/internal/wclayer"
+	"github.com/sirupsen/logrus"
+	"github.com/urfave/cli"
+)
+
+var (
+	debug = false
+	pause = false
+)
+
+func main() {
+
+	app := cli.NewApp()
+	cwd, _ := filepath.Abs("./")
+
+	app.Flags = []cli.Flag{
+		cli.StringFlag{
+			Name:  "i",
+			Usage: "Full path to .tar.gz of the root file system to be converted",
+			Value: cwd + `\rootfs.tar.gz`,
+		},
+		cli.StringFlag{
+			Name:  "o",
+			Usage: "Full path to output filename",
+			Value: cwd + `\rootfs.vhd`,
+		},
+		cli.IntFlag{
+			Name:  "s",
+			Usage: "Size in MB of the new VHD",
+			Value: 19,
+		},
+		cli.BoolFlag{
+			Name:  "D",
+			Usage: "Debug mode",
+		},
+		cli.BoolFlag{
+			Name:  "P",
+			Usage: "Pause when complete but before tearing down the utility VM",
+		},
+	}
+	app.Action = func(c *cli.Context) {
+		rootfs2vhd(c)
+	}
+	fmt.Printf("\nrootfs2vhd: Converts an LCOW root filesystem tar.gz to a VHD\n\n")
+	if err := app.Run(os.Args); err != nil {
+		os.Exit(-1)
+	}
+}
+
+func rootfs2vhd(c *cli.Context) {
+	sourceRootFS := c.String("i")
+	destFile := c.String("o")
+	debug = c.Bool("D")
+	pause = c.Bool("P")
+
+	if debug {
+		logrus.SetLevel(logrus.DebugLevel)
+		logrus.SetFormatter(&logrus.TextFormatter{FullTimestamp: true})
+	}
+
+	if _, err := os.Stat(sourceRootFS); err != nil {
+		fmt.Fprintf(os.Stderr, "%s not found\n", sourceRootFS)
+		os.Exit(-1)
+	}
+
+	if _, err := os.Stat(destFile); err == nil {
+		fmt.Fprintf(os.Stderr, "%s exists. Not overwriting\n", destFile)
+		os.Exit(-1)
+	}
+
+	fmt.Printf("- Creating %s...\n", destFile)
+	if _, err := exec.Command(`powershell`, `-command`, fmt.Sprintf(`New-VHD %s -SizeBytes %dMB -fixed`, destFile, c.Int("s"))).Output(); err != nil {
```

Comment:
This wont work on all machines. Should we do this via the storage api's instead?